analysis/README.md

```diff
@@ -4,15 +4,15 @@ This crate is where most of the compilation pipeline lives. It takes a parse tre
 
 ## Definitions
 
-We define two types of machines: virtual machines and constrained machines. Constrained machines are the lower level kind of machine. They have a notion of blocks through a latch and a function_id. Virtual machines are a higher level type of machines. For each machine type, we provide the elements which only appear exclusively in that type.
+We define two types of machines: virtual machines and constrained machines. Constrained machines are the lower level kind of machine. They have a notion of blocks through a latch and an operation_id. Virtual machines are a higher level type of machines. For each machine type, we provide the elements which only appear exclusively in that type.
 
 |                         | Virtual | Constrained |
 |-------------------------|---------|-------------|
 | pc                      | yes     | no          |
 | latch                   | no      | yes         |
-| function_id             | no      | yes         |
-| functions bodies        | yes     | no          |
-| internal instructions   | yes     | no          |
+| operation_id            | no      | yes         |
+| functions               | yes     | no          |
+| instructions            | yes     | no          |
```

Comment:
Good point but no, links are allowed in VMs

Reply:
```suggestion
| instructions            | yes     | no          |
| links                   | yes     | yes         |
```

This way then? ;) I think links should be mentioned here, otherwise it's the only major concept that's not.